Add tests for SQLJSPackageDB

mint-thompson opened the pull request:
Test the save, find, stats, and clear class methods. No tests are added for the logging methods.

Minor styling changes in some other files, as applied by prettier.

mint-thompson committed:
Add tests for SQLJSPackageDB

Test the save, find, stats, and clear class methods. No tests are added
for the logging methods.

Minor styling changes in some other files, as applied by prettier.

mint-thompson committed:
Merge branch 'new-fpl' into new-fpl-db-tests

cmoesel commented on test/db/SQLJSPackageDB.test.ts:
```diff
@@ -0,0 +1,589 @@
+          packageVersion: '3.2.2'
+        })
+      );
+      // ?? do we want to turn it back into a boolean, or is leaving it as a 0 good enough
```
What is this comment referencing? Is it perhaps related to how `sdAbstract` worked before you fixed it in the implementation?

mint-thompson replied:
Sure is! I've removed it.

cmoesel commented on test/db/SQLJSPackageDB.test.ts:
```diff
@@ -0,0 +1,589 @@
+
+    it('should return one resource when there is at least one match', () => {
+      const resource = packageDb.findResourceInfo('my-value-set');
+      expect(resource).toBeDefined();
```
Can you add another expectation to confirm that it returned the _right_ resource?

mint-thompson replied:
I've added an expectation that it matches one of the two resources that could be expected to be returned here.

cmoesel reviewed (changes requested):
This is quite nice. Elegant and clean. I only left a few comments -- nothing major. Thanks!

mint-thompson committed:
Make test for SQLJSPackageDB.findResourceInfo stricter

Check properties of the returned resource to make sure it matches one of
the expected results.

Remove unneeded comment from saveResourceInfo test.

cmoesel commented on test/db/SQLJSPackageDB.test.ts:
```diff
@@ -0,0 +1,600 @@
+    it('should return one resource when there is at least one match', () => {
+      const resource = packageDb.findResourceInfo('my-value-set');
+      expect(resource).toBeDefined();
+      // both valueSetThree and valueSetFour have a matching id,
+      // so either packageVersion is acceptable.
+      expect(resource).toEqual(
+        expect.objectContaining({
+          resourceType: 'ValueSet',
+          id: 'my-value-set',
+          url: 'http://example.org/ValueSets/my-value-set',
+          name: 'MyValueSet',
+          packageName: 'RegularPackage',
+          packageVersion: expect.stringMatching(/^(3\.2\.2|4\.5\.6)$/)
+        })
+      );
```
Technically, I think this should always return `valueSetFour` (version `4.5.6`) because we order by `rowid DESC`. This is intentional to match current `FHIRDefinitions` behavior (which is "last in wins").

cmoesel reviewed (changes requested):
Thank you! One more comment for your consideration.

mint-thompson committed:
Expect last added resource from findResourceInfo

cmoesel reviewed (approved):
Thanks, Mint. These tests looks great and cover all the needed functionality.
